main.go

```diff
@@ -72,6 +85,37 @@ func createPrincipalName(userArn arn.ARN) (string, error) {
 	}
 }
 
+func (c *config) validatePublicKey(sshPubKey ssh.PublicKey) error {
+	_, ok := sshPubKey.(ssh.CryptoPublicKey)
+	if !ok {
+		return fmt.Errorf("hallow: ssh public key is not a CryptoPublicKey")
+	}
+
+	pubKey := sshPubKey.(ssh.CryptoPublicKey).CryptoPublicKey()
+
+	switch pubKey.(type) {
+	case *rsa.PublicKey:
+		// rsa.PublicKey.Size() returns *bytes*. In an effort to make it easy
+		// to manage this list without doing math to figure out what the switch
+		// cases are, we're going to multiply this by 8 to get to *bits*, which
+		// is the number we all know and love.
+		size := (pubKey.(*rsa.PublicKey).Size() * 8)
+
+		switch size {
+		case 1024:
```

Comment:
Should we structure this an `if size < minKeySize`?

Reply:
Restructured!

Reply:
I wasn't sure if it was safe to do so, since I have no idea if it's possible to create a key with an awkwardly sized length above the minimum but containing bad properties. I think it's a silly concern, and someone doing something that wrong is likely doing worse stuff wrong too.